masonry/src/render_root.rs

```diff
@@ -570,8 +567,11 @@ impl RenderRoot {
             self.state.focused_widget = new;
             self.root_lifecycle(event);
 
-            // TODO - Handle IME
-            // Send TextFieldFocused(focused_widget) signal
+            self.state.signal_queue.push_back(if new.is_some() {
```

Comment:
It's not clear to me how you know that `new` is a textbox.

Reply:
No distinction is made between text boxes and other focused widgets, so all we know is that a widget is focused or that no widget is focused. 
Forgot that this was unaddressed.

Reply:
I've added a TODO so that this can be addressed in the next iteration of this (probably at the same time as introducing the Android/OSK IME changes).